x-pack/plugins/cases/server/client/configure/client.test.ts

```diff
@@ -346,6 +348,524 @@ describe('client', () => {
+                  category: null,
+                },
+              },
+            ],
+          },
+          version: 'test-version',
+        });
+
+        clientArgs.services.caseConfigureService.patch.mockResolvedValue({
+          id: 'test-id',
+          type: 'cases-configure',
+          version: 'test-version',
+          namespaces: ['default'],
+          references: [],
+          attributes: {
+            templates: [],
+            created_at: '2019-11-25T21:54:48.952Z',
+            created_by: {
+              full_name: 'elastic',
+              email: 'testemail@elastic.co',
+              username: 'elastic',
+            },
+            updated_at: '2019-11-25T21:54:48.952Z',
+            updated_by: {
+              full_name: 'elastic',
+              email: 'testemail@elastic.co',
+              username: 'elastic',
+            },
+          },
+        });
+
+        await expect(
+          update(
+            'test-id',
+            {
+              version: 'test-version',
+              templates: [],
+            },
+            clientArgs,
+            casesClientInternal
+          )
+        ).resolves.not.toThrow();
+      });
+
+      it(`throws when trying to update more than ${MAX_TEMPLATES_LENGTH} templates`, async () => {
+        await expect(
+          update(
+            'test-id',
+            {
+              version: 'test-version',
+              templates: new Array(MAX_TEMPLATES_LENGTH + 1).fill({
+                key: 'template_1',
+                name: 'template 1',
+                description: 'test',
+                caseFields: null,
+              }),
+            },
+            clientArgs,
+            casesClientInternal
+          )
+        ).rejects.toThrow(
+          `Failed to get patch configure in route: Error: The length of the field templates is too long. Array must be of length <= ${MAX_TEMPLATES_LENGTH}.`
+        );
+      });
+
+      it('throws when there are duplicated template keys in the request', async () => {
+        await expect(
+          update(
+            'test-id',
+            {
+              version: 'test-version',
+              templates: [
+                {
+                  key: 'template_1',
+                  name: 'template 1',
+                  description: 'test',
+                  caseFields: null,
+                },
+                {
+                  key: 'template_1',
+                  name: 'template 2',
+                  description: 'test',
+                  caseFields: {
+                    title: 'Case title',
+                  },
+                },
+              ],
+            },
+            clientArgs,
+            casesClientInternal
+          )
+        ).rejects.toThrow(
+          'Failed to get patch configure in route: Error: Invalid duplicated templates keys in request: template_1'
+        );
+      });
+
+      it('throws when there are invalid template keys in the request', async () => {
```

Comment:
Because `template_2` key doesn't exist in the configuration. this scenario verifies that we pass template key which matches with existing configuration. 
We do same for custom fields using `validateCustomFieldKeysAgainstConfiguration` function while creating a case, at line 52 `x-pack/plugins/cases/server/client/cases/create.ts`

Reply:
For Cases, it makes sense because we update the case SO. Here we update the configuration. Let's say I want to create a new template in the UI. How can I do that using the `PATCH` API?

Reply:
ahh.. true. It will not let me add new template due to this validation. 🤦‍♀️ 
I will remove it

Reply:
removed in https://github.com/elastic/kibana/pull/182575/commits/41deb1beeb129eb1849a1dad293de530190081cc